### ISSUE TITLE
test_live_upgrade_watchdog is flaky

### ISSUE THREAD

**Issue by rbradford** [cloud-hypervisor/cloud-hypervisor#4749]
This test has been observed as flaky

https://cloud-hypervisor-jenkins.westus.cloudapp.azure.com/job/cloud-hypervisor/job/main/1602/execution/node/278/log/
https://cloud-hypervisor-jenkins.westus.cloudapp.azure.com/job/cloud-hypervisor/job/main/1598/execution/node/148/log/

as well as PR builds

**Comment by likebreath** [comment]
Similar to what's being observed from PR #4748, likely the guest journal is somehow lost occasionally. 

Taking job1602 as example, the failed assertion suggested the guest does not reboot (as a result of watchdog) with an injected kernel crash on the migrated destination VM, while the CI log showed the opposite:
```
18:28:17  cloud-hypervisor: 169.612706612s: <__watchdog> ERROR:virtio-devices/src/watchdog.rs:173 -- Watchdog triggered: 33 seconds since last ping
18:28:17  cloud-hypervisor: 169.612906325s: <vmm> INFO:vmm/src/lib.rs:1686 -- VM reset event
```

**Comment by rbradford** [comment]
All we know is that `get_reboot_count()` is giving us unexpected numbers

```
fn get_reboot_count(guest: &Guest) -> u32 {
    // Ensure that the current boot journal is written so reboot counts are valid
    guest.ssh_command("sudo journalctl --sync").unwrap();

    guest
        .ssh_command("sudo journalctl --list-boots | wc -l")
        .unwrap()
        .trim()
        .parse::<u32>()
        .unwrap_or_default()
}
```

Of course, this doesn't actually check the number of reboots, just the number of lines of output on stdout of `--list-boots`. We could be getting a different message than we expect.

**Comment by likebreath** [comment]
I found the root-cause of the issue which is the systemd journal content was removed as it hits it size limit (10% of the underlining file system). For the case of our integration tests, the guest image is 2GB and hence the journal size is about 200MB. What's strange is that the journal size only grows rapidly on Azure VMs (reaching its limit easily) while the journal size stays around 24MB consistently on my local machine with the same set of integration tests on live-migration with watchdog.

The issue should be fixed by #4777 with an alternative way to count reboot times (which I believe is fairly stable). I would suggest running a periodic CI on this PR for a day to ensure it is stable.
